src/components/Listbox/components/Option/Option.tsx

```diff
@@ -89,20 +89,21 @@ export const Option = memo(function Option({
   return (
     <li
       {...sectionAttributes}
-      data-within-section={isWithinSection}
+      data-listbox-option
       data-listbox-option-value={value}
       data-listbox-option-destructive={destructive}
+      data-within-section={isWithinSection}
       className={classNames(styles.Option, divider && styles.divider)}
       id={domId}
       ref={listItemRef}
       tabIndex={-1}
-      onMouseDown={handleMouseDown}
-      aria-disabled={disabled}
-      onClick={disabled ? undefined : handleOptionClick}
       role={legacyRoleSupport}
       aria-label={accessibilityLabel}
       aria-selected={selected}
-      data-listbox-option
+      aria-disabled={disabled}
+      onClick={disabled ? undefined : handleOptionSelect}
+      onKeyPress={disabled ? undefined : handleOptionSelect}
```

Comment:
Should this be `onKeyUp`? I guess I can't really think of a real difference either way. I just normally use key up